jetty-core/jetty-util/src/test/java/org/eclipse/jetty/util/resource/JarResourceTest.java

```diff
@@ -234,25 +218,33 @@ public void testEncodedFileName()
     public void testJarFileResourceList() throws Exception
     {
         Path testJar = MavenTestingUtils.getTestResourcePathFile("jar-file-resource.jar");
-        URI uri = URI.create("jar:" + testJar.toUri().toASCIIString() + "!/");
+        URI uri = URIUtil.toJarFileUri(testJar.toUri());
         try (ResourceFactory.Closeable resourceFactory = ResourceFactory.closeable())
         {
             Resource resource = resourceFactory.newResource(uri);
             Resource rez = resource.resolve("rez/");
 
             assertThat("path /rez/ is a dir", rez.isDirectory(), is(true));
 
-            List<String> actual = rez.list();
-            String[] expected = new String[]{
-                "one",
-                "aaa",
-                "bbb",
-                "oddities/",
-                "another dir/",
-                "ccc",
-                "deep/",
-                };
-            assertThat("Dir contents", actual, containsInAnyOrder(expected));
+            try (Stream<Path> listStream = Files.list(rez.getPath()))
+            {
+                List<String> actual = listStream
+                    .map(Path::getFileName)
+                    .map(Path::toString)
+                    .sorted()
+                    .toList();
+
+                String[] expected = new String[]{
+                    "one",
+                    "aaa",
+                    "bbb",
+                    "oddities",
```

Comment:
I really think we should keep the trailing slashes, at least in this PR to avoid conflating multiple subjects.

Reply:
Incoming PR shortly